lib/membrane_rtmp_plugin/rtmp/source/message_handler.ex

```diff
@@ -0,0 +1,264 @@
+  end
+
+  defp do_handle_client_message(%Messages.CreateStream{tx_id: tx_id}, _header, state) do
+    stream_id = [1.0]
+
+    tx_id
+    |> Responses.default_result(stream_id)
+    |> send_rtmp_payload(state.socket, state.interceptor.chunk_size, chunk_stream_id: 3)
+
+    {:cont, state}
+  end
+
+  defp do_handle_client_message(
+         %Messages.Anonymous{name: "_checkbw", tx_id: tx_id},
+         _header,
+         state
+       ) do
+    tx_id
+    |> send_rtmp_payload(state.socket, state.interceptor.chunk_size, chunk_stream_id: 3)
+
+    {:cont, state}
+  end
+
+  defp do_handle_client_message(%Messages.Audio{data: data}, header, state) do
+    {buffers, state} = get_media_buffers(header, data, state)
+    {:cont, %{state | buffers: buffers}}
+  end
+
+  defp do_handle_client_message(%Messages.Video{data: data}, header, state) do
+    {buffers, state} = get_media_buffers(header, data, state)
+    {:cont, %{state | buffers: buffers}}
+  end
+
+  defp do_handle_client_message(%Messages.Anonymous{name: "deleteStream"}, _header, state) do
+    {:cont, state}
+  end
+
+  defp do_handle_client_message(%Messages.Anonymous{} = message, _header, state) do
+    Logger.debug("Unknown message: #{inspect(message)}")
+
+    {:cont, state}
+  end
+
+  defp request_packet(pid) do
+    send(pid, :need_more_data)
+  end
+
+  defp get_media_buffers(header, data, state) do
+    payload =
+      case state.header_sent? do
+        false ->
+          get_flv_header()
+
+        true ->
+          <<>>
+      end <> get_flv_body(header, data)
+
+    buffers = [%Buffer{payload: payload} | state[:buffers]]
+    {buffers, %{state | header_sent?: true}}
+  end
+
+  defp get_flv_header() do
+    <<"FLV", 0x01::8, 0::5, 1::1, 0::1, 1::1, 9::32>>
+  end
+
+  defp get_flv_body(
+         %Membrane.RTMP.Header{
+           #  chunk_stream_id: stream_id,
+           timestamp: timestamp,
+           body_size: data_size,
+           type_id: type_id,
+           stream_id: stream_id
+         },
+         payload
+       ) do
+    <<0::32, type_id::8, data_size::24, timestamp::24, 0::8, stream_id::24,
+      payload::binary-size(data_size)>>
+  end
+
+  defp send_rtmp_payload(message, socket, chunk_size, opts \\ []) do
+    type = Serializer.type(message)
+    body = Serializer.serialize(message)
+
+    chunk_stream_id = Keyword.get(opts, :chunk_stream_id, 2)
+
+    header =
+      [chunk_stream_id: 2, type_id: type, body_size: byte_size(body)]
+      |> Keyword.merge(opts)
+      |> Header.new()
+      |> Header.serialize()
+
+    payload = Message.chunk_payload(body, chunk_stream_id, chunk_size)
+
+    :gen_tcp.send(socket, [header | payload])
+  end
+
+  # The RTMP connection is based on TCP therefore we are operating on a continuous stream of bytes.
+  # In such case packets received on TCP sockets may contain a partial RTMP packet or several full packets.
+  #
+  # `Interceptor` is already able to request more data if packet is incomplete but it is not aware
```

Comment:
I'm wondering if it wouldn't be better if we operated on `recv` instead of the active pooling of messages. The we wouldn't have to care about those `need_more_data` and we would request as many bytes as we need.

But that shouldn't be done in that PR, rather a consideration for further improvement in the future.